src/keyboard/KeyboardViewport.tsx

```diff
@@ -0,0 +1,153 @@
+import { FC, PropsWithChildren, useEffect, useRef } from "react";
+
+type KeyboardViewportType = PropsWithChildren<{
+  className?: string;
+}>;
+
+const KEYMAP_SCALE = "keymap:scale";
+const DEFAULT_SCALE = window.localStorage.getItem(KEYMAP_SCALE) ?? "1";
+
+export const KeyboardViewport: FC<KeyboardViewportType> = ({
```

Comment:
I disagree that this simplifies the scale localstorage in any meaningful way. This ref management is hard to follow, and without the deserialization we have to keep casting back and forth between numbers and strings.

Reply:
this has been addressed as well.